packages/core/src/external/aws/lambda-logic/bundle-to-brief.ts

```diff
@@ -605,53 +814,62 @@ function replaceReferencesWithData(
   }
 
   if ("performer" in res) {
-    if (Array.isArray(res.performer)) {
+    if (res.performer) {
       if (
         res.resourceType === "DiagnosticReport" ||
         res.resourceType === "Observation" ||
         res.resourceType === "ServiceRequest"
       ) {
-        updRes.performer = res.performer
-          .flatMap(p => {
-            const ref = p.reference;
-            if (ref) {
-              referencedIds.add(ref);
-              const performer = map.get(ref);
-              if (
-                performer?.resourceType === "Organization" ||
-                performer?.resourceType === "Practitioner"
-              ) {
-                if (typeof performer.name === "string") return performer.name;
-                return getNameString(performer.name);
-              }
-            }
-            return [];
-          })
-          .join(", ");
+        const performers = toArray(res.performer);
+        const orgs: string[] = [];
+        const practitioners: string[] = [];
+
+        performers.forEach(p => {
+          const ref = p.reference;
+          if (!ref) return;
+
+          referencedIds.add(ref);
+          const performer = map.get(ref);
+
+          if (performer?.resourceType === "Organization") {
+            const name = typeof performer.name === "string" ? performer.name : "";
+            if (name.length > 0) orgs.push(name);
+          } else if (performer?.resourceType === "Practitioner") {
+            const name =
+              typeof performer.name === "string" ? performer.name : getNameString(performer.name);
```

Comment:
ditto: .trim() on .name

Reply:
actually, this check if it's a string is a left-over. Practitioner.name is never a string.